backend_modules/azure/host/main.tf

```
@@ -0,0 +1,240 @@
+  private_additional_security_group_id = var.base_configuration.private_additional_security_group_id
+  resource_group_name = var.base_configuration.resource_group_name
+  resource_name_prefix = "${var.base_configuration["name_prefix"]}${var.name}"
+  
+  public_instance = var.public_instance
+  
+  # availability_zone = var.base_configuration["availability_zone"]
+  region            = var.base_configuration["location"]
+}
+
+data "template_file" "user_data" {
+  count    = var.quantity > 0 ? var.quantity : 0
+  template = file("${path.module}/user_data.yaml")
+  vars = {
+    image           = var.image
+    public_instance = local.public_instance
+    mirror_url      = var.base_configuration["mirror"]
+  }
+}
+
+resource "azurerm_public_ip" "suma-pubIP" {
+  count = local.public_instance ? var.quantity: 0
+  name                = "${var.base_configuration["name_prefix"]}${var.name}-pubIP"
+  resource_group_name = local.resource_group_name
+  location            = local.region
+  allocation_method   = "Static"
+
+  tags = {
+    environment = "Production"
+  }
+}
+
+resource "azurerm_network_interface" "suma-main-nic" {
+  count = var.quantity
+  name                = "${var.base_configuration["name_prefix"]}${var.name}-main-nic${count.index}"
+  location            = local.region
+  resource_group_name = local.resource_group_name
+  
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = var.connect_to_base_network ? (local.public_instance ? local.public_subnet_id : local.private_subnet_id) : var.connect_to_additional_network ? local.private_additional_subnet_id : local.private_subnet_id
+    private_ip_address_allocation = "Dynamic"
+    public_ip_address_id = local.public_instance?azurerm_public_ip.suma-pubIP[count.index].id:null
+  }
+}
+
+resource "azurerm_network_interface" "suma-additional-nic" {
+  count = var.connect_to_base_network && var.connect_to_additional_network ? var.quantity : 0
+  name                = "${var.base_configuration["name_prefix"]}${var.name}-additional-nic${count.index}"
+  location            = local.region
+  resource_group_name = local.resource_group_name
+  
+
+  ip_configuration {
+    name                          = "internal-additional"
+    subnet_id                     =  local.private_additional_subnet_id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_linux_virtual_machine" "instance" {
+  count                            = var.quantity
+  name                             = "${local.resource_name_prefix}${var.quantity > 1 ? "-${count.index + 1}" : ""}"
+
+  location                         = local.region
+  resource_group_name              = local.resource_group_name
+  network_interface_ids            = compact(["${azurerm_network_interface.suma-main-nic[count.index].id}","${var.connect_to_additional_network?azurerm_network_interface.suma-additional-nic[count.index].id:""}"])
```

the check for the additional NIC can be done by calling the length function on the created resource, since the check to create the resource is different from the one defined in here.

```suggestion
  network_interface_ids            = compact(["${azurerm_network_interface.suma-main-nic[count.index].id}","${length(azurerm_network_interface.suma-additional-nic) > 0?azurerm_network_interface.suma-additional-nic[count.index].id:""}"])
```